Amazon/Marketplace/v7153/templates/jfrog-artifactory-ec2-instance.template.yaml

```diff
@@ -0,0 +1,408 @@
+      VPCZoneIdentifier: !Ref PrivateSubnetIds
+      MinSize: !Ref MinScalingNodes
+      MaxSize: !Ref MaxScalingNodes
+      Cooldown: '300'
+      DesiredCapacity: !Ref MinScalingNodes
+      TargetGroupARNs:
+        - !Ref TargetGroupARN
+        - !Ref SSLTargetGroupARN
+        - !Ref InternalTargetGroupARN
+      HealthCheckType: ELB
+      HealthCheckGracePeriod: 1800
+      Tags:
+        - Key: Name
+          Value: !Ref DeploymentTag
+          PropagateAtLaunch: true
+        - Key: ArtifactoryVersion
+          Value: !Ref ArtifactoryVersion
+          PropagateAtLaunch: true
+      TerminationPolicies:
+        - OldestInstance
+        - Default
+    CreationPolicy:
+      ResourceSignal:
+        Count: !Ref MinScalingNodes
+        Timeout: PT60M
+
+  ArtifactoryLaunchConfiguration:
+    Type: 'AWS::AutoScaling::LaunchConfiguration'
+    Metadata:
+      AWS::CloudFormation::Authentication:
+        S3AccessCreds:
+          type: S3
+          roleName:
+            - !Ref HostRole  # !Ref ArtifactoryHostRole
+          buckets:
+            - !Ref QsS3BucketName
+      AWS::CloudFormation::Init:
+        configSets:
+          artifactory_install:
+            - "config-cloudwatch"
+            - "config-artifactory-primary"
+            - "secure-artifactory"
+        config-cloudwatch:
+          files:
+            /root/cloudwatch.conf:
+              content: |
+                [general]
+                state_file = /var/awslogs/state/agent-state
+
+                [/var/log/messages]
+                file = /var/log/messages
+                log_group_name = /artifactory/instances/{instance_id}
+                log_stream_name = /var/log/messages/
+                datetime_format = %b %d %H:%M:%S
+
+                [/var/log/jfrog-ami-setup.log]
+                file = /var/log/messages
+                log_group_name = /artifactory/instances/{instance_id}
+                log_stream_name = /var/log/jfrog-ami-setup.log
+                datetime_format = %b %d %H:%M:%S
+
+                [/var/log/jfrog-ami-artifactory.log]
+                file = /var/log/messages
+                log_group_name = /artifactory/instances/{instance_id}
+                log_stream_name = /var/log/jfrog-ami-artifactory.log
+                datetime_format = %b %d %H:%M:%S
+              mode: "0400"
+        config-artifactory-primary:
+          files:
+            /root/attach_volume.sh:
+              content: !Sub |
+                #!/usr/bin/env bash
+                IS_PRIMARY="${ArtifactoryPrimary}"
+
+                if [[ $IS_PRIMARY != "true" ]]; then
+                  echo 'Not primary node. Skipping EBS volume attachment.'
+                  lsblk # debug
+                  exit 0
+                fi
+
+                echo "Using primary volume ID ${PrimaryVolume}"
+                VOLUME_ID="${PrimaryVolume}"
+                echo "VOLUME_ID: $VOLUME_ID"
+                if [[ -z "$VOLUME_ID" ]]; then
```

Comment:
one liner: `${VOLUME_ID:? Invalid volume id}`
This will
1. assert non-zero length
2. exit with error
3. Output your text
You can just remove the `if`